Add docs for experimental indexing features

kilfoyle opened the pull request:
Draft. Will open very soon.

ruflin commented on docs/en/ingest-management/data-streams.asciidoc:
```diff
@@ -445,3 +445,179 @@ The result should include `type: "boolean"` for the specified field.
+[[data-streams-experimental-tsds-disable]]
+== Disable TSDS
+
+To disable TSDS, follow the same procedure as to <<data-streams-experimental-tsds-enable,enable TSDS>>, but specify `null` for `index.mode` instead of `time_series`.
```
Lets add an example ehre that users can just copy / paste.

kilfoyle replied:
@ruflin Will do, but do you know where I can find that example? I don't have a test environment set up.

lalit-satapathy replied:
Here is an working example to disable TSDB on nginx.stubstatus.

https://github.com/elastic/ingest-docs/issues/151#issuecomment-1547447587

kilfoyle replied:
Thanks! I added the example as shown:

![Screenshot 2023-05-26 at 10 36 47 AM](https://github.com/elastic/ingest-docs/assets/41695641/f0293cc8-4319-4590-88eb-f618917457d0)

ruflin commented on docs/en/ingest-management/data-streams.asciidoc:
```diff
@@ -445,3 +445,179 @@ The result should include `type: "boolean"` for the specified field.
+
+[discrete]
+[[data-streams-experimental-tsds-enable]]
+== Enable TSDS
```
TSDS is GA, synthetic source is not. Lets put TSDS on the top.

kilfoyle replied:
I've moved TSDS above synthetic source.

andresrc commented:
AFAIK, as of 8.8.0 these features are no longer experimental. If this is correct, I'd probably go "Advanced Indexing Features" or something like that and remove the technical preview banner /cc @joshdover @ruflin

ruflin reviewed (approved):
I'm approving this change on my end. It is key that we have the page, now we can start to iterate on it with smaller PR's / changes.

kilfoyle commented:
@andresrc 
> AFAIK, as of 8.8.0 these features are no longer experimental...

Coincidentally, I worked on the TSDS docs. The feature is GA. I think all that matters is that the synthetic source docs (in the Elasticsearch guide) are marked with the admonition, and they are:
![Screenshot 2023-05-26 at 10 44 20 AM](https://github.com/elastic/ingest-docs/assets/41695641/5fcec350-9ee9-4d5e-959d-53e010693137)

So I've removed the technical preview banner from this page, and I've changed all instances of `experimental` to `advanced`.

lalit-satapathy commented on docs/en/ingest-management/data-streams.asciidoc:
```diff
@@ -445,3 +445,206 @@ The result should include `type: "boolean"` for the specified field.
+
+[source,sh]
+----
+POST metrics-nginx.stubstatus/_rollover
```
It should be "POST metrics-nginx.stubstatus-default/_rollover"

lalit-satapathy reviewed (approved):
Approving as the first cut version. We will add any tweaks needed in future docs PR.

kilfoyle commented:
Thanks for thorough and super fast reviews!

I'll merge and backport this into 8.8, and I'm happy to help iterate on whatever updates we still need.

kilfoyle commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.8|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/ingest-docs/228">](https://github.com/elastic/ingest-docs/pull/228)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->
